### DISCUSSION TITLE
Improve navigation with nnn

### DISCUSSION THREAD

**Discussion by jarun** [git-for-windows/git#4386] [Ideas]
Terminal manager [nnn](https://github.com/jarun/nnn) has a Cygwin port: https://cygwin.com/packages/summary/nnn.html
Is there a easy way to integrate it within git bash?

If not, would it be possible to package it as a navigation utility within git-bash?

**Comment by dscho** [comment]
The purpose of [Git for Windows](https://git-for-windows.github.io/) is really to bring *Git* to Windows.

While that implies that we have to ship quite a few Unix-like tools, it does not mean that we will include such tools unless they are needed for Git's own operations, or at least for historical reasons (needs of active contributors are sometimes a factor in deciding whether to include a tool or not).

Having said that, what you really are looking for is [MSYS2](https://msys2.github.io/). Git for Windows leverages MSYS2 and ships with a subset of its files. MSYS2 even sports a package management system called "pacman" to install more tools (including Git...), so why don't you give it a whirl?

**Reply by jarun** [reply]
Thanks for the insight.
I copied nnn.exe from MSYS2 environment to git-bash and it works out of the box.
